src/frontend/src/session.rs

```diff
@@ -974,6 +978,15 @@ impl SessionManager for SessionManagerImpl {
                             ),
                             salt,
                         }
+                    } else if auth_info.encryption_type == EncryptionType::Oauth as i32 {
+                        let reader = self
+                            .env
+                            .meta_client()
+                            .get_system_params()
+                            .await
+                            .map_err(|e| PsqlError::StartupError(e.into()))?;
+                        let oauth_jwks_url = reader.oauth_jwks_url().to_string();
```

Comment:
```suggestion
                        let oauth_jwks_url = self.env.system_params_manager.get_params().load().oauth_jwks_url().to_string();
```

We can read the jwks url directly from the parameter manager in env, then we don't have to fetch from meta for each session and keep the function non-async.